frontend/packages/operator-lifecycle-manager/integration-tests-cypress/views/operator.ts

```diff
@@ -0,0 +1,144 @@
+    installToSingleNamespace: boolean = false,
+  ) => {
+    cy.log(`install operator ${operatorName} in ${installToNamespace}`);
+    nav.sidenav.clickNavLink(['Operators', 'OperatorHub']);
+    cy.byTestID('search-operatorhub').type(operatorName);
+    cy.log('go to operator overview panel');
+    cy.byTestID(operatorHubCardID).click();
+    cy.log('go to the install form');
+    cy.byLegacyTestID('operator-install-btn').click({ force: true });
+    if (installToSingleNamespace) {
+      cy.log('configure Operator install form for single namespace');
+      cy.byTestID('A specific namespace on the cluster-radio-input').check();
+    } else {
+      cy.byTestID('All namespaces on the cluster-radio-input').should('be.checked');
+      // TODO some operators, such as 'Openshift Container Storage', do not support 'All namespaces' option and instead present a 'Operator recommended Namespace'
+      // cy.byTestID('Operator recommended Namespace:-radio-input').should('be.checked');
+    }
+    cy.log(`verify namespace dropdown shows the ${installToNamespace} namespace`);
+    cy.byTestID('dropdown-selectbox').should('contain', installToNamespace);
+
+    // Install
+    cy.byTestID('install-operator').click();
+    cy.log('verify Operator began installation');
+    cy.byTestID('view-installed-operators-btn').should(
+      'contain',
+      `View installed Operators in Namespace`,
+    );
+    cy.log(`navigate to OperatorHub in Namespace ${installToNamespace}`);
+    cy.byTestID('view-installed-operators-btn').click();
+    cy.log(`verify the Operator row for ${operatorName} exists`);
+    cy.byTestOperatorRow(operatorName, { timeout: 90000 }).should('exist');
+    cy.byTestOperatorRow(operatorName)
+      .parents('tr')
+      .within(() => {
+        cy.byTestID('status-text').should('have.text', 'Succeeded');
+      });
+  },
+  navToDetailsPage: (operatorName: string, installedNamespace: string) => {
+    cy.log(`navigate to details page of operator ${operatorName}`);
+    nav.sidenav.clickNavLink(['Operators', 'Installed Operators']);
+    projectDropdown.selectProject(installedNamespace);
+    projectDropdown.shouldContain(installedNamespace);
+    listPage.filter.byName(operatorName);
+    listPage.rows.countShouldBe(1);
+    cy.byTestOperatorRow(operatorName, { timeout: 60000 }).should('exist');
+    cy.byTestOperatorRow(operatorName).click();
+  },
+  uninstallModal: {
+    open: (operatorName: string, installedNamespace: string) => {
+      cy.log('open uninstall modal');
+      operator.navToDetailsPage(operatorName, installedNamespace);
+      detailsPage.clickPageActionFromDropdown('Uninstall Operator');
+      modal.shouldBeOpened();
+      modal.modalTitleShouldContain('Uninstall Operator?');
+      cy.get('.loading-skeleton--table').should('not.exist');
+    },
+    checkDeleteAllOperands: () =>
+      cy.byTestID('Delete all operand instances for this operator__checkbox').click(),
+  },
+  createInstance: (
```

Comment:
Separate CRUD action from assertion so that we aren't tied to a specific post-action state.
We should be able to do something like:

```
operator.createOperand(...)
operator.operandExists(...)
```

Reply:
We do assertions along-the-way as we do these OLM ops. 
For example for `operator.install()`
1. assert global or single  'installedNamespace` is in the ns dropdown on the install form
2. click Install button
3. assert 'Operator begun installation' messages
4. click 'view installed operator' button
5. assert operator appears in Installed Operators list
6. assert operator's status reaches 'Succeeded'.

I can't really take out/decouple the assertions from this flow.

Reply:
I just mean for the final resulting state. I think it's okay to assert certain states along the way, but I think it would be better if the `install` helper was agnostic to whether or not the installation should succeed or not.

Reply:
Same for the create action, it should be agnostic as to whether or not the creation was successful. That should be up to the test case to assert.

Reply:
Same for `operator.createInstance/Operand()`:
1. assert initial "No operands found" messages
2. nav to creation form
3. assert correct form title
4. type operand link name
5. click create button
6. assert link name exists in table
7. click on assert link
8. assert operand details page title

Reply:
> I just mean for the final resulting state. I think it's okay to assert certain states along the way...

Ok, I'll more out the last assertions from these commands, and add appropriate `shouldExist` helpers :-)